modules/library/main/src/test/java/org/geotools/filter/FilterTest.java

```diff
@@ -563,6 +567,26 @@ public void testLike() throws IllegalFilterException {
     }
 
     /**
+     * Tests the like operator with no-ascii texts.
+     *
+     * @throws IllegalFilterException If the constructed filter is not valid.
+     */
+    public void testUnicodeLike() throws IllegalFilterException {
+        
+        // Set up string
+        PropertyName testAttribute = new AttributeExpressionImpl(testSchema, "unicodeString");
+        
+        PropertyIsLike filter = fac.like(testAttribute, "*bara"+Character.toString((char)241)+"*", "*", ".", "!");
+        assertTrue(filter.evaluate(testFeature));
+        
+        filter = fac.like(testAttribute, "*BARA"+Character.toString((char)241).toUpperCase()+"*", "*", ".", "!");
+        assertTrue(filter.evaluate(testFeature));
+        
+        filter = fac.like(testAttribute, "*Bara"+Character.toString((char)241)+"*", "*", ".", "!", true);
+        assertTrue(filter.evaluate(testFeature));
```

Comment:
@ahuarte47 where are the negative tests? I want to see what happens to the first two tests if ```matchCase``` is ```true``` (I expect ```assertFalse```). You could even test all three with ```matchCase``` ```true``` and ```matchCase``` ```false```.

To make the test easier to read, and because ```FilterFactoryImpl`` makes the undocumented decision that ```matchCase``` defaults to ```false```, which is the opposite of OGC Filter, it might be clearer to always use an explicit ```matchCase```.